instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -48,37 +54,78 @@ def process(commands) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, M
                   s.record_exception(reply)
                   s.status = Trace::Status.error(reply.message)
                 end
+                next unless config[:record_value_size] && commands.length == 1
+
+                if RETRIEVED_VALUE_SIZE_COMMANDS.include?(commands[0][0])
+                  s['db.retrieved_value_size_bytes'] = calculate_bytesize(reply)
```

Comment:
IMO there's we should use the same attribute name for both `SET` and `RETRIEVE` cases - `db.value_size_bytes` seems reasonable (or `db.value_size` - similar semantic conventions are inconsistent).

Reply:
8147d43dc144003c5462f6b9b52bc67a230c6427

Reply:
Upon reflection, this makes telemetry on pipelined commands a little odd. Take e.g.:

```ruby
redis.pipelined do |p|
  p.set("X", "abc123")
  p.get("X")
end
```

If we have one attribute to store `SET`ted and `GET`ted values, then `span["value_size_bytes"]` would equal `("abc123" + "abc123").bytesize`. I think having `db.set_value_size_bytes` and `db.get_value_size_bytes`  could be more useful...

Reply:
🤔  I think we also will need to patch [`Redis::Client.call_pipelined`](https://github.com/redis/redis-rb/blob/master/lib/redis/client.rb#L214-L249) in order to access the response from pipelined calls 😅